### PR TITLE
feat: document and refactor GitHub example for readability

### PR DESCRIPTION
**Pull request by yuhgto**
Made some changes to make the Github example easier to understand and extend to another API. 

The changes:
- Split the custom trigger and custom action logic into their own files. Most developers start with either a trigger or action but not both.
- Documented all controller functions and most services.
- Explained what each DB table does.
- Cleaned up the Readme a bit

The documentation is written in JSDoc format so it will get picked up by most IDEs: 
Screen Shot 2023-03-10 at 3 06 32 PM

Let me know if you have any questions! :)

### CODE REVIEW

File: apps/github-node/src/controllers/trigger-controller.js [hunk 1]
```diff
@@ -46,42 +57,38 @@ async function unsubscribe(req, res) {
...
 
-async function executeAction(req, res) {
-  const { userId } = req.session;
-  const { inputFields } = req.body.payload;
+/**
+ * Trigger the integration recipe. 
+ * Receives events from a Github webhook and & calls the corresponding monday webhook URL. 
+ * Docs: https://developer.monday.com/apps/docs/custom-trigger#calling-your-action
+ */
+async function integrationEventsHandler(req, res) {
```
Comment (rongabbay) [inline apps/github-node/src/controllers/trigger-controller.js:65]: `triggerEventshHandler` maybe ?

**Review by rongabbay — APPROVED** [review]
looks very cool

did u try to simulate this on your local env ?

File: apps/github-node/src/controllers/trigger-controller.js (hunk 1 above)
Comment (yuhgto) [reply apps/github-node/src/controllers/trigger-controller.js:65]: nice, updated it

File: apps/github-node/src/controllers/action-controller.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+async function executeAction(req, res) {
+  const { userId } = req.session;
+  const { inputFields } = req.body.payload;
+
+  try {
+    // Retrieve the relevant user's OAuth token
+    const { token } = await connectionModelService.getConnectionByUserId(userId);
+
+    // Gets the blocks' required data from the input fields object.
+    // Input fields docs: https://developer.monday.com/apps/docs/custom-actions#configure-action-input-fields
```
[the remark this answers is not in the document]
Comment (yuhgto) [reply apps/github-node/src/controllers/action-controller.js:20]: good call, I explained it a bit more

File: apps/github-node/src/controllers/action-controller.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * Docs: https://developer.monday.com/apps/docs/custom-actions
+ * @param req The POST request from monday, which contains the block's input fields.
+ * @todo Update this to interact with your product's API instead of Github.
+ */
+async function executeAction(req, res) {
+  const { userId } = req.session;
+  const { inputFields } = req.body.payload;
+
+  try {
+    // Retrieve the relevant user's OAuth token
```
[the remark this answers is not in the document]
Comment (yuhgto) [reply apps/github-node/src/controllers/action-controller.js:16]: done!

File: apps/github-node/README.md [hunk 4]
```diff
@@ -15,26 +16,33 @@ This is the "Github Integration" example Monday app.
...
+- OAuth proccess to 3rd party
+
+### Extending the sample code
+
+The code is designed to be easy to understand and extend to any other API. We use JSDoc to document the main functions. There are `@todo` tags to explain some of the parts you need to adapt to your own API. 
 
 ## Install
 
-1. Make sure you have Node (v10+) and npm installed
+1. Make sure you have Node (v10+), npm , and [Node version manager](https://github.com/nvm-sh/nvm) installed
```
[the remark this answers is not in the document]
Comment (yuhgto) [reply apps/github-node/README.md:27]: This is not valid markdown

**Review by vladmonday — APPROVED** [review]

**Pull request merged**